src/trio/_core/_run.py

```diff
@@ -2953,7 +2953,12 @@ async def checkpoint_if_cancelled() -> None:
         _KqueueStatistics as IOStatistics,
     )
 else:  # pragma: no cover
-    raise NotImplementedError("unsupported platform")
+    if "eventlet" in sys.modules or "gevent" in sys.modules:
+        raise NotImplementedError(
+            "unsupported platform or primitives trio depends on are monkey-patched out",
+        )
+    else:
+        raise NotImplementedError("unsupported platform")
```

Comment:
maybe something like this?

```suggestion
    _patchers = sorted({"eventlet", "gevent"}.intersection(sys.modules.keys()))
    if _patchers:
        raise NotImplementedError(
            "unsupported platform or primitives trio depends on are monkey-patched out by {_patchers}",
        )
    else:
        raise NotImplementedError("unsupported platform")
```

Reply:
Else block is unnecessary after raise statement

Reply:
Ah of course